### PR TITLE
Replace *_Writer with write_fastx

### PR DESCRIPTION
**Pull request by standage**
Closes https://github.com/dib-lab/khmer/issues/1481.
- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality, is it tested?
- [x] `make format diff_pylint_report doc` Is it well formatted?
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by standage** [comment]
Ready for review @ctb @betatim @luizirber @camillescott

File: screed/screedRecord.py [hunk 1]
```diff
@@ -180,23 +180,31 @@ def _buildRecord(fieldTuple, dbObj, rowName, queryBy):
...
+            defline=defline,
+            sequence=record.sequence,
+            quality=record.quality)
+    else:
+        recstr = '>{defline}\n{sequence}\n'.format(
+            defline=defline,
+            sequence=record.sequence)
+
+    try:
+        fileobj.write(bytes(recstr, 'utf-8'))
```
Comment (betatim) [inline screed/screedRecord.py]: `f.write(recstr.encode('utf-8'))` should work in both versions of python and get rid of the need for `try ... except`. I think.

Comment (standage) [inline screed/screedRecord.py]: Get this now in Python 3.

```
>       fileobj.write(recstr.encode('utf-8'))
E       TypeError: string argument expected, got 'bytes'
```

Comment (betatim) [inline screed/screedRecord.py]: You need to use `open(fname, 'wb')` to write bytes.

**Comment by betatim** [comment]
LGTM. One more thing to consider: the callers now have to know to pass an `open(blah, "wb")` file object. How do we let them know about this? (or are the only users people reading this thread?)

**Comment by standage** [comment]
@titus, what do you think? Is it reasonable to require users to write in 'wb' mode?

**Comment by ctb** [comment]
yes, I think so. can we autodetect when it's in the wrong mode?

**Comment by betatim** [comment]
Yes we can!

```python
f = open('/tmp/sdfdsf', 'wb')
print(f.mode)
```

**Comment by standage** [comment]
Sweet! I'll hop on this...

**Comment by standage** [comment]
Ok, see latest commit. Feeling much better about this.

**Comment by betatim** [comment]
Merge then?

**Comment by standage** [comment]
Yes

**Comment by betatim** [comment]
🎉

**Pull request merged**